CycloneDX/Services/ProjectFileService.cs

```diff
@@ -270,11 +264,18 @@ public async Task<HashSet<DotnetDependency>> RecursivelyGetProjectDotnetDependen
                     project.Dependencies.Add(dependency.Name, dependency.Version);
                 }
 
-                dotnetDependencys.UnionWith(projectDotnetDependencys);
+                // A higher version might already be added
+                allAddedDepedencyNames = dotnetDependencys.Select(dep => dep.Name);
+                dotnetDependencys.UnionWith(projectDotnetDependencys.Where(pr => !allAddedDepedencyNames.Contains(pr.Name)).Select(d =>
+                {
+                    // Ensure project references are not added as direct references
+                    d.IsDirectReference = false;
+                    return d;
+                }));
```

Comment:
The current code would merge in older versions from referenced projects . In my case I got a second version of `Serilog` in which was incorrect.

`Serilog.Extensions.Hosting@8.0.0` on the referenced project used referenced `Serilog@3.1.1` but my main-project referenced already `Serilog@4.0.0` which was also set as dependency for `Serilog.Extensions.Hosting@8.0.0` in `dotnetDependencys` correctly. So `Serilog@3.1.1` was added as component but never referenced due to `Serilog.Extensions.Hosting@8.0.0` by `UnionWith` being taken from the main project.